File: eParking.Services/Database/Parking/Review.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eParking.Services.Database.Parking
{
    public class Review
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public MyAppUser User { get; set; } = null!;

        [Required]
        public int ParkingLotId { get; set; }

        [ForeignKey(nameof(ParkingLotId))]
        public ParkingLot ParkingLot { get; set; } = null!;

        [Required]
        public int Rating { get; set; } // 1-5

        [MaxLength(1000)]
        public string? Comment { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }
    }
}

File: UI/eParking_mobile/lib/models/review.dart
class Review {
  final int id;
  final int userId;
  final String userFullName;
  final int parkingLotId;
  final String parkingLotName;
  final int rating;
  final String? comment;
  final DateTime createdAt;

  Review({
    required this.id,
    required this.userId,
    required this.userFullName,
    required this.parkingLotId,
    required this.parkingLotName,
    required this.rating,
    required this.createdAt,
    this.comment,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'] as int,
      userId: json['userId'] as int,
      userFullName: json['userFullName'] as String? ?? '',
      parkingLotId: json['parkingLotId'] as int,
      parkingLotName: json['parkingLotName'] as String? ?? '',
      rating: json['rating'] as int,
      comment: json['comment'] as String?,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}
